src/conn.rs

```diff
@@ -620,14 +620,15 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
                         self.state = State::Closed { err: Some(err) };
                     } else {
                         let seq = *syn;
-                        *attempts += 1;
+                        let new_attempts = *attempts + 1;
+                        *attempts = new_attempts;
```

Comment:
It was to resolve this error:
```
error[E0502]: cannot borrow `*self` as immutable because it is also borrowed as mutable
   --> src/conn.rs:624:38
    |
612 |             State::Connecting(connected) => match &mut self.endpoint {
    |                                                   ------------------ mutable borrow occurs here
...
624 |                         let packet = self.syn_packet(seq);
    |                                      ^^^^^^^^^^^^^^^^^^^^ immutable borrow occurs here
...
629 |                         let timeout = self.config.initial_timeout * 2u32.pow(*attempts as u32);
    |                                                                              --------- mutable borrow later used here
```
Any alternative suggestions?

Reply:
huh, can we move the construction of the timeout above the construction of the packet?

if not, then I would rename `new_attempts` to `latest_attempt`

Reply:
Cool, yeah, looks like moving it up is an improvement: it has a nice benefit of avoiding a `packet.clone()`.